### PR TITLE
Apply better recursion pattern to `dss_untransform`

### PR DESCRIPTION
**Pull request by charleskawczynski**
#631 introduced a performance regression, this PR is an attempt to fix it.

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

File: src/Spaces/dss.jl [hunk 1]
```diff
@@ -146,6 +152,18 @@ end
     Geometry.transform(ax, targ, local_geometry)
 end
 
+if VERSION >= v"1.7.0"
```
Comment (simonbyrne) [inline src/Spaces/dss.jl:155]: I think we require v1.7 or greater

**Comment by simonbyrne** [comment]
Thanks, did this fix it?

**Comment by charleskawczynski** [comment]
> Thanks, did this fix it?

No, unfortunately, it's not. I'm fairly certain the issue is in the recursive `dss_untransform`. This runs but the allocations are actually slightly worse. I'm still looking into this.

**Comment by charleskawczynski** [comment]
Superseded by #691.

**Pull request closed without merging**